src/third/redux-persist/createPersistor.js

```diff
@@ -49,13 +49,18 @@ export default function createPersistor (store, config) {
         updatedSubstates.push([key, state[key]])
       });
 
+      const writes = []
       while (updatedSubstates.length > 0) {
         await new Promise(r => setTimeout(r, 0));
 
         const [key, substate] = updatedSubstates.shift()
-        const storageKey = createStorageKey(key)
-        storage.setItem(storageKey, serializer(substate)).catch(warnIfSetError(key))
+        writes.push([key, serializer(substate)])
       }
+
+      writes.forEach(([key, serializedSubstate]) => {
+        storage.setItem(createStorageKey(key), serializedSubstate).catch(warnIfSetError(key))
+      })
+
```

Comment:
createPersistor [nfc]: Separate serializing from writing.

This is definitely not NFC -- it's basically the main behavior improvement this series makes :wink: It should narrow the race window quite a lot.

The `multiSet` is a nice bonus, which probably narrows the race window further by making just one trip from JS over to the native side. But this change already means that the `setItem` calls are no longer separated by (a) the CPU time to do the serialization, or (b) yields.

Reply:
Separately, it'd be good to have a yield between the last serialization and doing the writes.

I see that happens later in the series (by moving the yield to the bottom instead of the top of the loop body). That's fine, then, because that bit isn't critical.